model/postgres/postgres_server.rb

```diff
@@ -106,6 +112,20 @@ def configure_hash
   end
 
   def trigger_failover
+    DB.transaction do
+      if read_replica? && resource.representative_server.id == id && (target = read_replica_failover_target)
+        update(representative_at: nil)
+        target.update(representative_at: Time.now)
+        resource.incr_refresh_dns_record
+        return true
+      end
+
+      if read_replica?
+        Clog.emit("Failed to trigger read replica failover")
+        return false
+      end
```

Comment:
nit: a single branch for read replicas might be easier to read
```suggestion
      if read_replica?
        if resource.representative_server.id == id && (target = read_replica_failover_target)
          update(representative_at: nil)
          target.update(representative_at: Time.now)
          resource.incr_refresh_dns_record
          return true
        end
        Clog.emit("Failed to trigger read replica failover")
        return false
      end
```